mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -264,7 +276,8 @@ class Word2Vec extends Serializable with Logging {
 
   /**
    * Computes the vector representation of each word in vocabulary.
-   * @param dataset an RDD of words
+   * @param dataset a RDD of sentences,
+   *                each sentence is expressed as an iterable collection of words
```

Comment:
I think we should remove L286 and change `learnVocab` to `learnVocab(dataset)`. The `val words = dataset.flatMap(x => x)` can the be moved into `learnVocab`. This seems clearer to me, to avoid confusion, since `words` is no longer used apart from the learn vocab step.